ballerina/channel_csv_io.bal

```diff
@@ -40,30 +40,90 @@ Error {
     return (check getReadableCSVChannel(readableChannel, 0)).csvStream();
 }
 
-isolated function channelWriteCsv(WritableChannel writableChannel, string[][]|map<anydata>[] content) returns Error? {
-    WritableCSVChannel csvChannel = check getWritableCSVChannel(writableChannel);
-    if content is string[][] {
-        foreach string[] r in content {
+isolated function channelWriteCsv(string path, FileWriteOption option, string[][]|map<anydata>[] contentToWrite) returns Error? {
+
+    WritableCSVChannel csvChannel;
+    if contentToWrite is string[][] {
+        csvChannel = check getWritableCSVChannel(check openWritableCsvFile(path, option = option));
+        foreach string[] r in contentToWrite {
             Error? writeResult = csvChannel.write(r);
             if writeResult is Error {
                 check csvChannel.close();
                 return writeResult;
             }
         }
-    } else if content is map<anydata>[] {
-        foreach map<anydata> row in content {
+        check csvChannel.close();
+    } else if contentToWrite is map<anydata>[] {
+        string[] headers = [];
+        if contentToWrite.length() == 0 {
+            return;
+        }
+        if option == OVERWRITE {
+            headers = contentToWrite[0].keys();
```

Comment:
shall we move outside the if clause, as this is needed in both scenario

Reply:
updated in [9967638](https://github.com/ballerina-platform/module-ballerina-io/pull/827/commits/9967638b794331da447a8be1d99326691ec4e69f)